experimental/LieAlgebras/src/CoxeterGroup.jl

```diff
@@ -0,0 +1,3 @@
+abstract type CoxeterGroup end # <: Group
```

Comment:
Why doesn't this subtype `Group`?

Reply:
GroupsCore requires `order` to be implemented, but since a `WeylGroup` can be constructed from an arbitrary Cartan matrix, we would have to detect the type to return the order. Thus, I left this for later.